core/app/models/spree/shipping_method.rb

```diff
@@ -30,10 +30,18 @@ class ShippingMethod < Spree::Base
     has_many :shipping_method_stock_locations, dependent: :destroy, class_name: "Spree::ShippingMethodStockLocation"
     has_many :stock_locations, through: :shipping_method_stock_locations
 
+    has_many :store_shipping_methods, inverse_of: :shipping_method
+    has_many :stores, through: :store_shipping_methods
+
     validates :name, presence: true
 
     validate :at_least_one_shipping_category
 
+    scope :available_to_store, ->(store) do
+      raise ArgumentError, "You must provide a store" if store.nil?
+      store.shipping_methods.empty? ? all : where(id: store.shipping_method_ids)
```

Comment:
This is the opposite of the direction we usually have the "none means all" rule.

Normally we would have "shipping method with no store is available to all stores" not "a store with no shipping methods can use all shpiping methods"

Reply:
Apparently payment methods do the same

https://github.com/solidusio/solidus/blob/master/core/app/models/spree/payment_method.rb#L39-L42

I'm not really a fan of that either

Reply:
This is the direction we use for most of these types of associations. (As you found with payment methods.)

I agree that it's confusing and I'm also not a fan of the "none means all" pattern we've adopted, but it's probably best to at least keep them consistent.